### PR TITLE
ENT-1467: Make the deterministic JDK image compatible with IntelliJ.

### PR DESCRIPTION
**Pull request by chrisr3**
Put enough content into the `jdk8u-deterministic/jdk` directory to fool IntelliJ into believing it to be a valid JDK image.

Also fix Gradle to ensure that we always use latest the `deterministic-rt.jar` file from the repository.

### CODE REVIEW

File: jdk8u-deterministic/build.gradle [hunk 1]
```diff
@@ -8,24 +8,43 @@ repositories {
...
 ext {
-    jdk_home = "$projectDir/jdk"
+    jdk_home = "$projectDir/jdk".toString()
     rt_jar = "$jdk_home/jre/lib/rt.jar".toString()
 }
 
 configurations {
-    jdk
+    jdk.resolutionStrategy {
+        cacheChangingModulesFor 0, 'seconds'
```
Comment (exFalso) [inline jdk8u-deterministic/build.gradle]: Doesn't this cause gradle to roundtrip to all repositories every time it's invoked?

Comment (chrisr3) [reply jdk8u-deterministic/build.gradle]: I'm hoping it triggers a `HEAD` request to determine whether something is the latest SNAPSHOT.

Comment (exFalso) [reply jdk8u-deterministic/build.gradle]: In my experience the up-to-date checks of gradle are quite dumb, it literally goes over all repositories and tries different combinations of artifacts. We used to use this in my prev workplace and it made every gradle invocation incredibly slow (although we had many more modules). Can we set this to a couple of hours/1 day perhaps?

Comment (chrisr3) [reply jdk8u-deterministic/build.gradle]: This triggers these 3 update-checks for every build:
```
Cached resource https://ci-artifactory.corda.r3cev.com/artifactory/corda-dev/net/corda/deterministic-rt/1.0-SNAPSHOT/maven-metadata.xml is up-to-date (lastModified: Mon Jun 25 13:10:54 BST 2018).
Cached resource https://ci-artifactory.corda.r3cev.com/artifactory/corda-dev/net/corda/deterministic-rt/1.0-SNAPSHOT/deterministic-rt-1.0-20180625.120901-7.pom is up-to-date (lastModified: Mon Jun 25 13:10:54 BST 2018).
Cached resource https://ci-artifactory.corda.r3cev.com/artifactory/corda-dev/net/corda/deterministic-rt/1.0-SNAPSHOT/deterministic-rt-1.0-20180625.120901-7-api.jar is up-to-date (lastModified: Mon Jun 25 13:09:34 BST 2018).
```
But I don't see any evidence that it has polled every single repository to do it.

Comment (exFalso) [reply jdk8u-deterministic/build.gradle]: Ah ok, good to go then!

**Pull request merged**